app/src/main/java/org/mozilla/focus/activity/MainActivity.kt

```diff
@@ -158,11 +159,14 @@ class MainActivity : LocaleAwareAppCompatActivity(), OnUrlEnteredListener {
     override fun dispatchKeyEvent(event: KeyEvent): Boolean {
         val fragmentManager = supportFragmentManager
         val browserFragment = fragmentManager.findFragmentByTag(BrowserFragment.FRAGMENT_TAG) as BrowserFragment?
+        val homeFragment = fragmentManager.findFragmentByTag(HomeFragment.FRAGMENT_TAG) as HomeFragment?
 
-        return if (browserFragment == null || !browserFragment.isVisible) {
-            super.dispatchKeyEvent(event)
+        return if (browserFragment != null && browserFragment.isVisible) {
+            browserFragment.dispatchKeyEvent(event)
+        } else if (homeFragment != null && homeFragment.isVisible) {
+            homeFragment.dispatchKeyEvent(event)
         } else {
-            browserFragment.dispatchKeyEvent(event) || super.dispatchKeyEvent(event)
```

Comment:
See deleted code here: if browserfragment doesn't handle the code, we want this to be handled by the super.

Try to be more careful when moving behavior around - removing the `super` call would have introduced regressions.